test/unit/services/NotificationServiceSpec.scala

```diff
@@ -18,84 +18,169 @@ package services
 
-      val result = await(service.save(SuccessNotification))
+      when(mockRepository.save(any())).thenReturn(Future.successful(Left(exception)))
+
+      val result = await(service.parseAndSave(SuccessNotificationXml))
 
       result mustBe Left(exception)
     }
 
     "return notification if exists based on the reference" in {
-      val notification = Notification("e4d94295-52b1-4837-bdc0-7ab8d7b0f1af", "SUCCESS", "sample.pdf")
-
-      when(mockRepository.find(any())(any())).thenReturn(Future.successful(List(notification)))
+      when(mockRepository.findNotificationsByReference(any())).thenReturn(Future.successful(List(parsedNotification)))
 
-      val result = await(service.findNotificationByReference("e4d94295-52b1-4837-bdc0-7ab8d7b0f1af"))
+      val result = await(service.getNotificationForReference(fileReference))
 
-      result mustBe Some(notification)
+      result mustBe Some(parsedNotification)
     }
 
     "return None if notification doesn't exist" in {
 
-      when(mockRepository.find(any())(any())).thenReturn(Future.successful(List.empty))
+      when(mockRepository.findNotificationsByReference(any())).thenReturn(Future.successful(List.empty))
 
-      val result = await(service.findNotificationByReference("reference"))
+      val result = await(service.getNotificationForReference("reference"))
 
       result mustBe None
     }
   }
+
+  "NotificationService on reattemptParsingUnparsedNotifications" should {
+    "do nothing if no unparsed notifications exist" in {
+      when(mockRepository.findUnparsedNotifications())
+        .thenReturn(Future.successful(Seq()))
+
+      await(service.reattemptParsingUnparsedNotifications())
+
+      verify(mockRepository, times(0)).updateNotification(any())
+    }
+
+    "reparse single unparsed notification that still can not be parsed" in {
+      when(mockRepository.findUnparsedNotifications())
+        .thenReturn(Future.successful(Seq(unParsedNotification.copy(payload = "<Root></Root>"))))
+
+      await(service.reattemptParsingUnparsedNotifications())
+
+      verify(mockRepository, times(0)).updateNotification(any())
+    }
+
+    "reparse single unparsed notification that can now be parsed" in {
+      when(mockRepository.findUnparsedNotifications())
+        .thenReturn(Future.successful(Seq(unParsedNotification)))
+
+      await(service.reattemptParsingUnparsedNotifications())
+
+      verify(mockRepository, times(1)).updateNotification(any())
+    }
+  }
+
+  "NotificationService on parseNotificationsPayload" should {
+    "return a parsed Notification when all required element are present" in {
+      val payload = <Root>
+        <FileReference>e4d94295-52b1-4837-bdc0-7ab8d7b0f1af</FileReference>
+        <BatchId>1</BatchId>
+        <FileName>sample.pdf</FileName>
+        <Outcome>SUCCESS</Outcome>
+        <Details>[detail block]</Details>
+      </Root>
```

Comment:
Nit: Could these be extracted into a single value? Referenced by different test cases?

Reply:
Lol, already had this in TestData class. Fixed